tests/hla/process_randomlist.py

```diff
@@ -0,0 +1,204 @@
+                        numQualSuccess += 1
+                    # Unsuccessful datasets
+                    else:
+                        print("TEST_RANDOM. Unsuccessful Dataset (fit_qual = 5): ", dataset, "\n")
+                        numUnsuccessful += 1
+
+                # Append the latest dataset table to the summary table 
+                allDatasetTable = vstack([allDatasetTable, datasetTable])
+
+            # Catch anything that happens as this dataset will be considered a failure, but
+            # the processing of datasets should continue.  Generate sufficient output exception
+            # information so problems can be addressed.
+            except Exception:
+           
+                exc_type, exc_value, exc_tb = sys.exc_info()
+                traceback.print_exception(exc_type, exc_value, exc_tb, file=sys.stdout)
+                print("TEST_RANDOM. Exception Dataset: ", dataset, "\n")
+                numException += 1
+                continue
+
+        # Perform some clean up
+        if os.path.isfile('ref_cat.ecsv'): 
+            os.remove('ref_cat.ecsv')
+        if os.path.isfile('refcatalog.cat'):  
+            os.remove('refcatalog.cat')
+        for filename in os.listdir():
+            if filename.endswith('flt.fits') or filename.endswith('flc.fits'):
+                os.unlink(filename)
+
+        # Write out the table
+        allDatasetTable.write('resultsBigTest.ecsv', format='ascii.ecsv')
+
+        # Determine the percent success over all datasets processed
+        percentSuccess = numSuccess/numProcessedDatasets
+        print('TEST_RANDOM. Number of tests started: ', numStartTests)
+        print('TEST_RANDOM. Number of tests (excluding filtered): ', numProcessedDatasets)
+        print('TEST_RANDOM. Number of successful tests: ', numSuccess)
+        print('TEST_RANDOM. Number of qualified successful tests: ', numQualSuccess)
+        print('TEST_RANDOM. Number of unsuccessful tests: ', numUnsuccessful)
+        print('TEST_RANDOM. Number of exception tests: ', numException)
+        print('TEST_RANDOM. Percentage success/numberOfTests: ', numSuccess/numProcessedDatasets*100.0)
+        print('TEST_RANDOM. Percentage success+qualsuccess/numberOfTests: ', (numSuccess+numQualSuccess)/numProcessedDatasets*100.0)
+ 
+        return percentSuccess
+
+def get_dataset_list(tableName):
+    """ Standalone function to read the Astropy table and get the dataset names
+
+    Parameters
+    ==========
+    tableName : str
+        Filename of the input master CSV file containing individual
+        images or association names, as well as observational
+        information regarding the images
+
+    Returns
+    =======
+    datasetNames: list
+        List of individual image or association base (IPPSSOOT) names
+    """
+
+    #dataFromTable = Table.read(filename, format='ascii')
+    datasetIDs = tableName['observationID']
```

Comment:
Since these variables are not used anywhere else, it would be simpler to simply use the column names directly in the code below, making the definitions for datasetIDs and asnIDs un-necessary.

Reply:
Agreed - fixed.